### ISSUE TITLE
 TypeError: Object 0 has no method 'toLowerCase' on Windows

### ISSUE THREAD

**Issue by martijnwalraven** [npm/node-which#16]
On Windows, `which()` always fails with `TypeError: Object 0 has no method 'toLowerCase'` in the case when `process.env.PATH` isn't defined (but `process.env.Path` is).

Code was added in https://github.com/isaacs/node-which/commit/ff956d87fce80cfa373065d0cb328a5aaed27900 to deal with case sensitivity of environment variables under Windows, but there is a bug in this code path where `p.toLowerCase()` should be `k[p].toLowerCase()` (and `process.env[p]` should also be changed accordingly).

```javascript
  // On windows, env.Path is common.
  if (isWindows && !pathEnv) {
    var k = Object.keys(process.env)
    for (var p = 0; p < k.length; p++) {
      if (p.toLowerCase() === 'path') {
        pathEnv = process.env[p]
        break
      }
    }
  }
```
https://github.com/isaacs/node-which/blob/master/which.js#L46-L55

**Comment by Oregu** [comment]
The fix was proposed in #15

**Comment by isaacs** [comment]
Fixed by @zkat
